### ISSUE TITLE
Suggestion: add a property called shouldRender to renderMathInElement options

### ISSUE THREAD

**Issue by benstevens48** [Khan/KaTeX#691]
I would like to suggest adding an optional property called `shouldRender` to the `renderMathInElement` options which if provided would accept the current HTML element being considered and return a Boolean indicating whether or not to render math inside it. This would be used in place of `ignoreTags`. Alternatively you could make `ignoreTags` optionally a function in the same way but the name makes less sense. This would neatly cover any scenario for opting out of math rendering and users could come up with their own conventions (like putting the attribute `data-render-math="false"` on elements they don't want rendered).

**Comment by edemaine** [comment]
Neat idea.  Maybe `ignoredTags` could accept a function, and we also allow a parallel-named option like `renderTags` which could also accept a list or function (with the intent of only passing in one of the two).

This will have to wait until #656 is complete, though.

**Comment by benstevens48** [comment]
Yes, as long as the function gets passed the HTML element being considered so can look at the attributes then I don't mind. I'm not sure that having renderTags as well as ignoredTags would add much - you would really need access to the attributes I think unless you were using custom elements (which is possible I suppose). E.g. if you put 'div' in renderTags, that's quite general! But, yes I suppose you might define a <cm-math> element in which case a list of renderTags could make sense.

**Comment by an-42-an** [comment]
Hi, I was looking at this issue and noticed an earlier PR (#1666) that was closed without merging. I wanted to confirm whether it was closed due to the lack of support for including both `shouldRender` and `ignoredTags`. If that is the case, and the same value is provided in both of these attributes, which one is expected to take precedence? If this is open to contribution, I would like to work on this issue following the preferred approach.
